test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java

```diff
@@ -127,6 +127,10 @@
     public static final String CLIENT_SOCKET_TIMEOUT = "client.socket.timeout";
     public static final String CLIENT_PATH_PREFIX = "client.path.prefix";
 
+    // This set will contain the warnings already asserted since we are eliminating logging duplicate warnings.
+    // This ensures that no matter in what order the tests run, the warning is asserted once.
+    private static Set<String> assertedWarnings = new HashSet<>();
```

Comment:
Is this potentially introducing a race condition if tests run concurrently (or if a test just happens to create multiple threads that log concurrently) due to the non-synchronized de-duplication implemenation?

Reply:
Using ConcurrentHashMap keySet here.